docs/eks.md

```diff
@@ -0,0 +1,343 @@
+        snapshotter = "soci"
+        # This line is required for containerd to send information about how to lazily load the image to the snapshotter
+        disable_snapshot_annotations = false
+EOF
+```
+
+
+### Step 4: Create SOCI install script
+
+This will create an script that will be run on instance boot to install and configure the SOCI snapshotter.
+
+```
+cat <<'EOF_SCRIPT' >install_soci.sh
+#!/bin/bash
+# Set environment variables
+ARCH=$(uname -m | sed s/aarch64/arm64/ | sed s/x86_64/amd64/)
+VERSION=0.7.0
+ARCHIVE=soci-snapshotter-$VERSION-linux-$ARCH.tar.gz
+
+pushd /tmp
+# Download, verify, and install the soci-snapshotter
+curl --silent --location --fail --output $ARCHIVE https://github.com/awslabs/soci-snapshotter/releases/download/v$VERSION/$ARCHIVE
+curl --silent --location --fail --output $ARCHIVE.sha256sum https://github.com/awslabs/soci-snapshotter/releases/download/v$VERSION/$ARCHIVE.sha256sum
+sha256sum ./$ARCHIVE.sha256sum
+tar xzvf ./$ARCHIVE -C /usr/local/bin soci-snapshotter-grpc
+rm ./$ARCHIVE
+rm ./$ARCHIVE.sha256sum
+
+# Configure the SOCI snapshotter for CRI credentials
+mkdir -p /etc/soci-snapshotter-grpc
+cat <<EOF >/etc/soci-snapshotter-grpc/config.toml
+[cri_keychain]
+# This tells the soci-snapshotter to act as a proxy ImageService
+# and to cache credentials from requests to pull images.
+enable_keychain = true
+# This tells the soci-snapshotter where containerd's ImageService is located.
+image_service_path = "/run/containerd/containerd.sock"
+EOF
+
+# Start the soci-snapshotter
+curl --silent --location --fail --output /etc/systemd/system/soci-snapshotter.service https://raw.githubusercontent.com/awslabs/soci-snapshotter/v$VERSION/soci-snapshotter.service
+systemctl daemon-reload
+systemctl enable --now soci-snapshotter
+
+popd
+EOF_SCRIPT
+```
+
+
+### Step 5: Create EC2 Userdata
+
+This step will combine the node configuration and SOCI installation script from the previous steps into a MIME multipart archive that works as EC2 user data. 
+
+
+```
+cat <<EOF > userdata.txt
+MIME-Version: 1.0
+Content-Type: multipart/mixed; boundary="BOUNDARY"
+
+--BOUNDARY
+Content-Type: application/node.eks.aws
+
+---
+$(cat node_config.yaml)
+
+--BOUNDARY
+Content-Type: text/x-shellscript; charset="us-ascii"
+
+$(cat install_soci.sh)
+
+--BOUNDARY--
+EOF
+```
+
+### Step 6: Create a node launch template
+
+Next, we will create a launch template that uses our chosen AMI, instance type, and userdata. This will be used to launch new nodes for our EKS cluster.
+
+```
+cat <<EOF > launch_template_data.json
+{
+  "ImageId": "${AMI_ID}",
+  "InstanceType": "${INSTANCE_TYPE}",
+  "UserData": "$(cat userdata.txt | base64 --wrap=0)"
+}
+EOF
+
+aws ec2 create-launch-template \
+  --launch-template-name soci-eks-node \
+  --launch-template-data file://launch_template_data.json \
+  --region $AWS_REGION
+```
+
+After creating the launch template, we can retrieve it's ID into a variable.
```

Comment:
its*

Reply:
I'd like to get this in. I think we can fix this in a follow up that includes more debugging information.